Fix new cli hammer options

ntkathole opened the pull request:
Issue #5224 

Test results

```
pytest tests/foreman/cli/test_hammer.py
=============================================================== test session starts ===============================================================
collected 1 item s
tests/foreman/cli/test_hammer.py .

============================================================ 1 passed in 7.32 seconds =====================================================
```


```
E           AssertionError:
E           hammer bootdisk
E             Added subcommands:
E               * subnet
E
E           hammer bootdisk subnet (new command)
E             Added options:
E               * subnet
E               * force
E               * help
E               * sudo
E               * subnet-id
E               * file
```

https://bugzilla.redhat.com/show_bug.cgi?id=1498454 bugzilla added new options for bootdisk subnet.

```
E           hammer host create
E             Added options:
E               *  -id

E           hammer host update
E             Added options:
E               * openscap-proxy-id
```
https://bugzilla.redhat.com/show_bug.cgi?id=1505295 bugzilla added option for openscap-proxy while creating/updating host
```
E
E           hammer host-collection hosts
E             Added options:
E               * thin

E
E           hammer host list
E             Removed options:
E               * include
E

```
https://bugzilla.redhat.com/show_bug.cgi?id=1503586  Remove --include and --thin options


```
E           hammer repository-set disable
E             Removed options:
E               * new-name
E
E           hammer repository-set enable
E             Removed options:
E               * new-name
```

https://bugzilla.redhat.com/show_bug.cgi?id=1501991 bugzilla removed unused option.
```
E
E           hammer user
E             Added subcommands:
E               * ssh-keys

E
E           hammer user ssh-keys (new command)
E             Added subcommands:
E               * info
E               * add
E               * list
E               * delete
E             Added options:
E               * help
E
E           hammer user ssh-keys add (new command)
E             Added options:
E               * name
E               * user
E               * key
E               * key-file
E               * user-id
E               * help
E
E           hammer user ssh-keys delete (new command)
E             Added options:
E               * id
E               * user-id
E               * user
E               * name
E               * help
E
E           hammer user ssh-keys info (new command)
E             Added options:
E               * id
E               * user-id
E               * user
E               * name
E               * help
E
E           hammer user ssh-keys list (new command)
E             Added options:
E               * search
E               * help
E               * page
E               * user
E               * user-id
E               * order
E               * per-page
E
```

https://bugzilla.redhat.com/show_bug.cgi?id=1476224 SSH Key in User Template functionality added in CLI

```

E           hammer user update
E             Added options:
E               * role-ids
E               * roles

E
E           hammer user create
E             Added options:
E               * role-ids
E               * roles

```

https://bugzilla.redhat.com/show_bug.cgi?id=1489324 Functionality of Assigning role to the user is added

```
E
E           hammer file (new command)
E             Added subcommands:
E               * info
E               * list
E             Added options:
E               * help
E
E           hammer file info (new command)
E             Added options:
E               * content-view-id
E               * product
E               * product-id
E               * name
E               * repository
E               * content-view
E               * organization-label
E               * content-view-version
E               * organization-id
E               * content-view-version-id
E               * organization
E               * repository-id
E               * id
E               * help
E
E           hammer file list (new command)
E             Added options:
E               * product-id
E               * help
E               * content-view-version
E               * by
E               * environment
E               * repository-id
E               * product
E               * repository
E               * content-view-filter-id
E               * content-view-id
E               * organization-id
E               * content-view-version-id
E               * environment-id
E               * content-view
E               * search
E               * organization-label
E               * content-view-filter
E               * ids
E               * order
E               * full-result
E               * organization
E               * page
E               * per-page
E
```
https://bugzilla.redhat.com/show_bug.cgi?id=1306736 is regarding new feature in 6.3 

```
E
E           hammer job-template
E             Added subcommands:
E               * import
E               * export
E
E           hammer job-template export (new command)
E             Added options:
E               * help
E               * name
E               * id
E
E           hammer job-template import (new command)
E             Added options:
E               * file
E               * overwrite
E               * help
E
```
https://bugzilla.redhat.com/show_bug.cgi?id=1457614 describes import and export functionality for templates

```
E           hammer report delete
E             Removed options:
E               * name
E
E           hammer report info
E             Removed options:
E               * name
E
E           hammer job-invocation info
E             Removed options:
E               * name
E
E           hammer capsule content synchronize
E             Added options:
E               * skip-metadata-check

E           hammer proxy content synchronize
E             Added options:
E               * skip-metadata-check
```

I could not find BZ for these options but these changes seems valid.

abalakh commented:
> I could not find BZ for these options but these changes seems valid.

`skip-metadata-check` sounds valid for me too, but are you sure removing `name` option for _some_ report and job-invocation subcommands is expected? That's how regression usually looks like, so I'd like to see some proofs these changes are intended.

abalakh commented:
btw pls try to fix your commit. 40k lines of changes in [diff](https://github.com/SatelliteQE/robottelo/commit/696437b39e3a76290613ee9460db966314d09823.diff) is not cool :-/

ntkathole commented:
@abalakh I fixed my commit and now the PR contains only those changes for which I mentioned bugzillas. Also I added changes for `skip-metadata-check ` option in capsule. Now test results looks like

```
            )
E           AssertionError:
E           hammer job-invocation info
E             Removed options:
E               * name
E
E           hammer report delete
E             Removed options:
E               * name
E
E           hammer report info
E             Removed options:
E               * name

tests/foreman/cli/test_hammer.py:156: AssertionError
```
I know the test will still fail. But it helps for automation analysis.
